src/model.py

```diff
@@ -1927,29 +1992,70 @@ def init_components(self, subtasks):
         #######################################
         # Task-specific CNN layers
         #######################################
-        for task in self.subtasks:
-            components[task].update(
-                {"cnn": torch.nn.Sequential(*[
-                    cnn_block(cnn_dim, cnn_dim)
-                    for layer in range(self.find(task+"_layers", default=1))
-                ])
-            })
+        # NOTE: stack can be 0, then these components are skipped. # TODO test this implementation
+        for stack in range(stack_count):
+            for task in self.subtasks:
+                task_layers = self.find(task+"_layers", default=1)
+
+                # handle three different cnn types for subtasks
+                if expanding_cnn:  # Sometimesse will be 0, other times None
+                    components[task].update(
+                        {f"cnn_{stack}": torch.nn.Sequential(*[
+                            self.expanding_cnn_block(cnn_dim, cnn_dim, kernel_size, m=expanding_cnn)
+                            for layer in range(task_layers)
+                            ])}
+                        )
+                elif None not in (split_cnn_tasks, split_cnn_kernels): 
+                    if task in split_cnn_tasks:
+                        components[task].update({
+                            f"cnn_{stack}": self.split_cnn_block(
+                                cnn_dim, cnn_dim, split_cnn_kernels, task_layers
+                                )
+                            })
```

Comment:
Both `split_cnn_kernels` and `split_cnn_tasks`  have to be present in order to add task wise split cnns. If they are not provided, then regular cnn blocks are used.